### PR TITLE
Express router boilerplate

### PR DESCRIPTION
**Pull request by spraguejdev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,143 @@
+# hrxp-api
```
Comment (miken619) [inline README.md:1]: Can you add a TODO to finalize this doc?

File: README.md [hunk 2]
```diff
@@ -1 +1,143 @@
...
+
+- [Getting Started](#getting-started)
+- [Database Tables](#database-tables)
+  - [Schema](#schema)
+- [Endpoints](#endpoints)
+- [Data requests and responses](#Data-requests-and-responses)
+
+# Getting Started
+
+- Run `npm i` from the command line.
```
Comment (miken619) [inline README.md]: Can you add a description of what these two CLI do?

File: README.md [hunk 3]
```diff
@@ -1 +1,143 @@
...
+
+Structure of request:
+
+```
+{
+
+}
+```
+
+**HTTP Status:**
```
Comment (miken619) [inline README.md:78]: I think the `HTTP Status` can be group together, unless you're planning on having different fields for each request.

File: README.md [hunk 4]
```diff
@@ -1 +1,143 @@
...
+| []()                       | GET    |             |           |
+
+# Data requests and responses
+
+Below are all expected request data responses.
+
+## `/channels`
+
+**Method:** GET
+
```
Comment (miken619) [inline README.md:69]: Description of what this `GET` request does please, same for the rest of the request.

File: db/index.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+// const mongoose = require('mongoose');
```
Comment (miken619) [inline db/index.js:1]: I don't think we should have commented out code. It just open doors and we can end up with a codebase where half our code is commented out and eventually it becomes clutter with unnecessary data. 

Can you add a commented out TODO instead? Same for the rest of your code.

File: db/model/channel.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// const db = require("../index");
+
+module.exports = {
+  example: async () => {
```
Comment (miken619) [inline db/model/channel.js]: Can you move mock code into a test module?

File: server/index.js [hunk 7]
```diff
@@ -1,16 +1,20 @@
-const express = require("express");
-const morgan = require("morgan");
-const cors = require("cors");
+const express = require('express');
+const morgan = require('morgan');
+const cors = require('cors');
 const app = express();
+const channel = require('./controller/channel.js');
 // Middleware
 app.use(express.json());
```
Comment (miken619) [inline server/index.js:7]: Can you add a description of what each of the middleware does?

File: server/index.js [hunk 8]
```diff
@@ -1,16 +1,20 @@
...
-app.use(morgan("dev"));
+app.use(morgan('dev'));
 app.use(cors());
 
-app.get("/", (req, res) => {
-  res.send("hello world");
+app.get('/', (req, res) => {
+  res.send('example endpoint');
```
Comment (miken619) [inline server/index.js:12]: Please move mock code into a test module.

**Review by miken619 — COMMENTED** [review]
Please add tests.

File: server/index.js (hunk 8 above)
Comment (shockey) [reply server/index.js:12]: this is OK IMO, it's per the spec in [the ticket for this work](https://trello.com/c/pWfygajy/8-architecture-set-up-api-boilerplate)

File: db/index.js (hunk 5 above)
Comment (shockey) [reply db/index.js:1]: Agree, let's either leave a `// TODO: ` comment or remove this altogether.

File: server/index.js (hunk 8 above)
Comment (miken619) [reply server/index.js:12]: I see, ignored my last suggestion.

File: README.md (hunk 1 above)
Comment (spraguejdev) [reply README.md:1]: Kyle is going to take care of the documentation. He is going to integrate Swaggerhub into our repo sometime soon!

File: README.md (hunk 2 above)
Comment (spraguejdev) [reply README.md]: Valid point!

File: README.md (hunk 3 above)
Comment (spraguejdev) [reply README.md:78]: Kyle is going to take care of the api documentation. But I see what you are saying

File: README.md (hunk 4 above)
Comment (spraguejdev) [reply README.md:69]: Done!

File: db/index.js (hunk 5 above)
Comment (spraguejdev) [reply db/index.js:1]: Good idea!

File: server/index.js (hunk 7 above)
Comment (spraguejdev) [reply server/index.js:7]: Yup!

**Comment by spraguejdev** [comment]
Hey Mike! Thanks for the detailed feedback! Those changes are implemented and I also will add test's in a later PR. 

Happy Friday!

File: db/index.js (hunk 5 above)
Comment (miken619) [reply db/index.js]: can you submit your new changes, I'm still seeing your commented out code.

**Pull request merged**